src/terminal/parser/ut_parser/OutputEngineTest.cpp

```diff
@@ -689,133 +689,141 @@ class StatefulDispatch final : public TermDispatch
         return true;
     }
 
-    bool CursorHorizontalPositionAbsolute(_In_ size_t const uiPosition) override
+    bool CursorHorizontalPositionAbsolute(_In_ size_t const uiPosition) noexcept override
     {
         _cursorHorizontalPositionAbsolute = true;
         _cursorDistance = uiPosition;
         return true;
     }
 
-    bool VerticalLinePositionAbsolute(_In_ size_t const uiPosition) override
+    bool VerticalLinePositionAbsolute(_In_ size_t const uiPosition) noexcept override
     {
         _verticalLinePositionAbsolute = true;
         _cursorDistance = uiPosition;
         return true;
     }
 
-    bool CursorPosition(_In_ size_t const uiLine, _In_ size_t const uiColumn) override
+    bool CursorPosition(_In_ size_t const uiLine, _In_ size_t const uiColumn) noexcept override
     {
         _cursorPosition = true;
         _line = uiLine;
         _column = uiColumn;
         return true;
     }
 
-    bool CursorSaveState() override
+    bool CursorSaveState() noexcept override
     {
         _cursorSave = true;
         return true;
     }
 
-    bool CursorRestoreState() override
+    bool CursorRestoreState() noexcept override
     {
         _cursorLoad = true;
         return true;
     }
 
-    bool EraseInDisplay(const DispatchTypes::EraseType eraseType) override
+    bool EraseInDisplay(const DispatchTypes::EraseType eraseType) noexcept override
     {
         _eraseDisplay = true;
         _eraseType = eraseType;
         return true;
     }
 
-    bool EraseInLine(const DispatchTypes::EraseType eraseType) override
+    bool EraseInLine(const DispatchTypes::EraseType eraseType) noexcept override
     {
         _eraseLine = true;
         _eraseType = eraseType;
         return true;
     }
 
-    bool InsertCharacter(_In_ size_t const uiCount) override
+    bool InsertCharacter(_In_ size_t const uiCount) noexcept override
     {
         _insertCharacter = true;
         _cursorDistance = uiCount;
         return true;
     }
 
-    bool DeleteCharacter(_In_ size_t const uiCount) override
+    bool DeleteCharacter(_In_ size_t const uiCount) noexcept override
     {
         _deleteCharacter = true;
         _cursorDistance = uiCount;
         return true;
     }
 
-    bool CursorVisibility(const bool fIsVisible) override
+    bool CursorVisibility(const bool fIsVisible) noexcept override
     {
         _cursorVisible = fIsVisible;
         return true;
     }
 
-    bool SetGraphicsRendition(const std::basic_string_view<DispatchTypes::GraphicsOptions> options) override
+    bool SetGraphicsRendition(const std::basic_string_view<DispatchTypes::GraphicsOptions> options) noexcept override
     {
-        _options.assign(options.cbegin(), options.cend());
-        _setGraphics = true;
+        try
+        {
+            _options.assign(options.cbegin(), options.cend());
+            _setGraphics = true;
 
-        return true;
+            return true;
+        }
+        catch (...)
```

Comment:
nit: more compact representation:

```
try
{
thing
return true;
}
CATCH_LOG()
return false;
```

(instead of doing to catch/log/return dance manually?)

Reply:
I made a CATCH_LOG_RETURN_FALSE() friendo.